x-pack/plugins/reporting/server/export_types/printable_pdf/execute_job/index.ts

```diff
@@ -27,15 +27,30 @@ export const runTaskFnFactory: RunTaskFnFactory<RunTaskFn<TaskPayloadPDF>> =
 
       const process$: Rx.Observable<TaskRunResult> = Rx.of(1).pipe(
         mergeMap(() => decryptJobHeaders(encryptionKey, job.headers, jobLogger)),
-        mergeMap((headers) => getCustomLogo(reporting, headers, job.spaceId, jobLogger)),
+        mergeMap(async (headers) => {
+          const fakeRequest = reporting.getFakeRequest(headers, job.spaceId, jobLogger);
+          const uiSettingsClient = await reporting.getUiSettingsClient(fakeRequest);
+          return getCustomLogo(uiSettingsClient, headers);
+        }),
         mergeMap(({ headers, logo }) => {
-          const urls = getFullUrls(reporting, job);
+          const urls = getFullUrls(reporting.getServerInfo(), reporting.getConfig(), job);
 
           const { browserTimezone, layout, title } = job;
           apmGetAssets?.end();
 
           apmGeneratePdf = apmTrans?.startSpan('generate-pdf-pipeline', 'execute');
-          return generatePdfObservable(reporting, {
+          //  make a new function that will call reporting.getScreenshots
+          const snapshotFn = () =>
+            reporting.getScreenshots({
```

Comment:
It's OK to leave reportingCore in the `execute_job` / `create_job` functions for now. We'll sweep through these in the next PR.

Reply:
I just wanted to make sure I wasn't misinterpreting this - so how this file and others are is ok, right? I had refactored the generatePdfObservable() to take in the snapshot function vs all of reporting core. 

Eventually I will remove any call to reporting.getScreenshots() etc in this file and others in future PRs :yay: but for this PR, this can stay as it is right?

Reply:
Yes, these files called `x-pack/plugins/reporting/server/export_types/*/execute_job*` as well as `x-pack/plugins/reporting/server/export_types/csv_v2/create_job.ts` are OK, and how I would expect them to be at this stage of our progress.

Reply:
Ok thank you!!